### PR TITLE
1777/feat/duplicate ids validation

### PR DESCRIPTION
**Pull request by y-golde**
also fixes 1779,

see ticket for reference 

### re-wrore id validation schema
cleaned, moved to switch form and added an optional test param (in this case to determine whether the id already exists in the investigaton
### added duplicate ids in same form validation
see ticker 1779 for more info

### CODE REVIEW

**Comment by y-golde** [comment]
Roll version after merge

**Review by barnahum8 — DISMISSED** [review]

File: client/src/Utils/Contacts/ContactIdValidationSchema.ts [hunk 1]
```diff
@@ -5,26 +5,36 @@ import InteractionEventContactFields from 'models/enums/InteractionsEventDialogC
...
         }
+
+        return schema;
     });
 
+interface TestParam {
+    name : string;
+    errorMsg : string;
+    testingFunction: (param : any) => boolean
+}
```
Comment (Noam6421) [inline client/src/Utils/Contacts/ContactIdValidationSchema.ts:38]: ;

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionDialog.tsx [hunk 2]
```diff
@@ -40,12 +40,20 @@ const InteractionDialog = (props: Props) => {
...
-    const { alertWarning } = useCustomSwal();
+    const { alertWarning, alertError } = useCustomSwal();
+
+    const getEventContactIds = () => {
+        const ids = interactions.map(interaction => interaction.contacts).flat().map((contact) => 
+                contact[InteractionEventContactFields.IDENTIFICATION_NUMBER]
+        );
+        return ids;
+    };
+
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionDialog.tsx:51]: delete row

**Review by Noam6421 — CHANGES_REQUESTED** [review]

**Comment by Noam6421** [comment]
GREAT PR

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionDialog.tsx (hunk 2 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionDialog.tsx:51]: done

**Review by Noam6421 — APPROVED** [review]

**Review by barnahum8 — APPROVED** [review]

File: client/src/Utils/Contacts/ContactIdValidationSchema.ts (hunk 1 above)
Comment (y-golde) [reply client/src/Utils/Contacts/ContactIdValidationSchema.ts]: done

**Pull request merged**